src/Pager.js

```diff
@@ -0,0 +1,282 @@
+        set(state.finished, 0),
+        set(state.time, 0),
+        set(this._index, index),
+        startClock(this._clock),
+      ]),
+      // Animate the values with a spring
+      spring(this._clock, state, config),
+      cond(state.finished, [
+        // When spring animation finishes, stop the clock
+        stopClock(this._clock),
+        // Reset gesture and velocity from previous gesture
+        set(this._gestureX, 0),
+        set(this._velocityX, 0),
+        call([this._index], ([value]) => {
+          // If the index changed, and previous spring was finished, update state
+          const route = this.props.navigationState.routes[Math.round(value)];
+
+          this.props.jumpTo(route.key);
+        }),
+      ]),
+    ]);
+  };
+
+  handleGestureEvent = event([
+    {
+      nativeEvent: {
+        translationX: this._gestureX,
+        velocityX: this._velocityX,
+        state: this._gestureState,
+      },
+    },
+  ]);
+
+  translateX = block([
+    onChange(
+      // Index changed from outside
+      this._nextIndex,
+      cond(or(eq(this._index, this._nextIndex), this._isSwipeGesture), NOOP, [
+        // Stop any running animations
+        stopClock(this._clock),
+        // Update the index to trigger the transition
+        set(this._index, this._nextIndex),
+      ])
+    ),
+    cond(
+      eq(this._gestureState, State.ACTIVE),
+      [
+        cond(this._isSwiping, NOOP, [
+          // We weren't dragging before, set it to true
+          set(this._isSwiping, TRUE),
+          set(this._isSwipeGesture, TRUE),
+          // Also update the drag offset to the last position
+          set(this._offsetX, this._position),
+        ]),
+        // Update position with previous offset + gesture distance
+        set(this._position, add(this._offsetX, this._gestureX)),
+        // Stop animations while we're dragging
+        stopClock(this._clock),
+      ],
+      [
+        set(this._isSwiping, FALSE),
+        this.transitionTo(
+          // Calculate the next index
+          cond(
+            or(
+              greaterThan(abs(this._gestureX), this._swipeDistanceThreshold),
+              greaterThan(abs(this._velocityX), this._swipeVelocityThreshold)
+            ),
+            // For swipe gesture, to calculate the index, determine direction and add to index
+            round(
+              min(
+                max(
+                  0,
+                  sub(
+                    this._index,
+                    cond(
+                      greaterThan(
+                        // Gesture can be positive, or negative
+                        // Get absolute for comparision
+                        abs(this._gestureX),
+                        this._swipeDistanceThreshold
+                      ),
+                      // If gesture value exceeded the threshold, calculate direction from distance
+                      divide(this._gestureX, abs(this._gestureX)),
```

Comment:
What's the reason for making it so hacky? Dividing on vary small integers in Java may give some unwanted effects (heard on Uni xd) and it's a bit more expensive. Of course, it probably does not matter, but if I were you I would use `cond(greatedThan(_gestureX, 0), 1, -1)`

Reply:
I copied it from my previous impl where it is JS. Will change it :)